File: bridgeAddon/addon/globalPlugins/nvdaMcpBridge/domain/controllers/teardown_reason.py
# nvdaMcpBridge domain -- TeardownReason: why a session ended.
# Copyright (C) 2026 Marlon Brandao de Sousa. GPL-2. See COPYING.txt.
#
# ROLE: a domain-only enum (plain Enum, per AGENTS.md -- it never crosses the
# wire, so it is not a StrEnum in protocol.py).
# USED BY: the Session controller (sets it on every exit path), the
# SessionContext's close() capability (a command asks to end the session with a
# reason), and the Transcript (the value is the human string on SESSION CLOSE).
#
# It lives in its own file so both session.py and commands/session_context.py can
# import it without a cycle (the context's close() takes a reason; the Session
# owns the teardown that consumes it).

from __future__ import annotations

import enum


class TeardownReason(enum.Enum):
	"""Why a session ended; the value is the transcript's SESSION CLOSE string."""

	CLIENT_BYE = "client-bye"
	CHANNEL_CLOSED = "channel-closed"
	HEARTBEAT_TIMEOUT = "heartbeat-timeout"
	INACTIVITY_TIMEOUT = "inactivity-timeout"
	HANDSHAKE_FAILED = "handshake-failed"
	EXTERNAL = "external"

File: bridges/voiceover/Sources/VoiceOverBridgeDomain/Entities/TeardownReason.swift
// ROLE: entity -- every way a session can end.
// The raw value is the string the transcript's session close line carries.
// USED BY: the Session on every exit path, the `bye` handler and BridgeServer.
public enum TeardownReason: String, Equatable, Sendable {
	case clientBye = "client-bye"

	case channelClosed = "channel-closed"

	case heartbeatTimeout = "heartbeat-timeout"

	case inactivityTimeout = "inactivity-timeout"

	case handshakeFailed = "handshake-failed"

	case external
}
